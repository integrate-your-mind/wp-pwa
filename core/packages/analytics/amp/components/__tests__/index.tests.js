import React from 'react';
import renderer from 'react-test-renderer';
import { types } from 'mobx-state-tree';
import { Provider } from 'mobx-react';
import Analytics from '..';

const Stores = types.model('Stores').props({
  build: types.frozen({
    isAmp: true,
    channel: 'amp',
    packages: 'saturn-theme,wp-org-connection',
  }),
  connection: types.frozen({
    selectedItem: { type: 'post', id: 60 },
  }),
  analytics: types.frozen({
    googleAnalytics: {
      ids: ['UA-12345678-1', 'UA-12345678-2'],
      trackingOptions: id => ({
        sendPageViews: id !== 'UA-12345678-1',
        sendEvents: true,
      }),
      pageViewAmp: {
        title: 'The Beauties of Gullfoss â€“ Demo Frontity',
        documentLocation:
          'https://demo.frontity.test/the-beauties-of-gullfoss/amp/',
        extraUrlParams: {
          cd1: 'dim1_post60',
          cd2: 'dim2_post60',
        },
      },
      ampVars: {},
      ampTriggers: {
        openMenu: {
          on: 'click',
          selector: '.menu',
          request: 'event',
          vars: {
            eventCategory: 'AMP - Post bar',
            eventAction: 'AMP - open menu',
          },
        },
      },
    },
    googleTagManager: {
      ids: ['GTM-112233'],
      ampVars: {
        anonymize: false,
        siteId: 'site1122334455',
        userIds: ['user00', 'user01'],
        theme: 'saturn-theme',
        extensions: 'saturn-theme,wp-org-connection',
        plan: 'enterprise',
        pageType: 'amp',
        dev: true,
        site: 'https://demo.frontity.test',
        title: 'The Beauties of Gullfoss',
        url: 'https://demo.frontity.test/the-beauties-of-gullfoss/amp/',
        type: 'post',
        id: 60,
        format: 'amp',
        route: 'single',
        hash: 'ZO0H2I9kJ0arOdstZdG',
        customDimensions: {
          dimension1: 'dim1_post60',
          dimension2: 'dim2_post60',
        },
      },
    },
    comScore: {
      ids: ['test1', 'test2'],
    },
  }),
});

describe('Analytics > AMP Components', () => {
  it('renders correctly', () => {
    const stores = Stores.create();
    const tree = renderer.create(
      <Provider stores={stores}>
        <Analytics />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
