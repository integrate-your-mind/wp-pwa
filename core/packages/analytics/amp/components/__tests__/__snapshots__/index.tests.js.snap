// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics > AMP Components renders correctly 1`] = `
Array [
  <amp-analytics
    type="googleanalytics"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{
  \\"vars\\": {
    \\"account\\": \\"UA-12345678-1\\"
  },
  \\"triggers\\": {
    \\"openMenu\\": {
      \\"on\\": \\"click\\",
      \\"selector\\": \\".menu\\",
      \\"request\\": \\"event\\",
      \\"vars\\": {
        \\"eventCategory\\": \\"AMP - Post bar\\",
        \\"eventAction\\": \\"AMP - open menu\\"
      }
    }
  }
}",
        }
      }
      type="application/json"
    />
  </amp-analytics>,
  <amp-analytics
    type="googleanalytics"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{
  \\"vars\\": {
    \\"account\\": \\"UA-12345678-2\\"
  },
  \\"triggers\\": {
    \\"openMenu\\": {
      \\"on\\": \\"click\\",
      \\"selector\\": \\".menu\\",
      \\"request\\": \\"event\\",
      \\"vars\\": {
        \\"eventCategory\\": \\"AMP - Post bar\\",
        \\"eventAction\\": \\"AMP - open menu\\"
      }
    },
    \\"trackPageview\\": {
      \\"on\\": \\"visible\\",
      \\"request\\": \\"pageview\\",
      \\"vars\\": {
        \\"title\\": \\"The Beauties of Gullfoss â€“ Demo Frontity\\",
        \\"documentLocation\\": \\"https://demo.frontity.test/the-beauties-of-gullfoss/amp/\\"
      }
    }
  },
  \\"extraUrlParams\\": {
    \\"cd1\\": \\"dim1_post60\\",
    \\"cd2\\": \\"dim2_post60\\"
  }
}",
        }
      }
      type="application/json"
    />
  </amp-analytics>,
  <amp-analytics
    config="https://www.googletagmanager.com/amp.json?id=GTM-112233&gtm.url=SOURCE_URL"
    data-credentials="include"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{
  \\"vars\\": {
    \\"anonymize\\": false,
    \\"siteId\\": \\"site1122334455\\",
    \\"userIds\\": [
      \\"user00\\",
      \\"user01\\"
    ],
    \\"theme\\": \\"saturn-theme\\",
    \\"extensions\\": \\"saturn-theme,wp-org-connection\\",
    \\"plan\\": \\"enterprise\\",
    \\"pageType\\": \\"amp\\",
    \\"dev\\": true,
    \\"site\\": \\"https://demo.frontity.test\\",
    \\"title\\": \\"The Beauties of Gullfoss\\",
    \\"url\\": \\"https://demo.frontity.test/the-beauties-of-gullfoss/amp/\\",
    \\"type\\": \\"post\\",
    \\"id\\": 60,
    \\"format\\": \\"amp\\",
    \\"route\\": \\"single\\",
    \\"hash\\": \\"ZO0H2I9kJ0arOdstZdG\\",
    \\"customDimensions\\": {
      \\"dimension1\\": \\"dim1_post60\\",
      \\"dimension2\\": \\"dim2_post60\\"
    }
  }
}",
        }
      }
      type="application/json"
    />
  </amp-analytics>,
  <amp-analytics
    type="comscore"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"vars\\":{\\"c2\\":\\"test1\\"}}",
        }
      }
      type="application/json"
    />
  </amp-analytics>,
  <amp-analytics
    type="comscore"
  >
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "{\\"vars\\":{\\"c2\\":\\"test2\\"}}",
        }
      }
      type="application/json"
    />
  </amp-analytics>,
]
`;
